src/main/java/pl/edu/zut/mad/schedule/ScheduleService.java

```diff
@@ -54,9 +54,6 @@ public ScheduleService(GroupAlbumRepository groupAlbumRepository,
     }
 
     List<Schedule> findBy(Map<String, String> params) {
-        if (scheduleRepository.count() == 0) {
```

Comment:
This condition was a cause of low performance?

Reply:
This was contributing to low performance along with some other factors. This method was called quite often and having two calls to database inside sometimes caused connection not closing properly. 

Since theres no real disadvantage of returning empty list I have removed this call as it wasn't really necessary.